lib/types.js

```diff
@@ -52,24 +52,13 @@
  * @typedef {Buffer} Block
  */
 
+// TODO: Figure out where those extra fields come from and find more elegant way to represent this
 /**
- * @typedef {Object} NoiseSecretStream
- * @property {Buffer} remotePublicKey
- * @property {number} remotePort
- * @property {string} remoteHost
- * @property {import('net').Socket} rawStream
- * @property {function} end
+ * @typedef {import('streamx').Duplex & {remoteAddress: string, remotePort:number}} RawDHTConnectionStream
```

Comment:
Seth may have an idea of where these come from, but my guess is that it would require even more work to get the proper typings for that to update this

Reply:
i think the approach used here makes sense.